engine/src/main/java/org/hibernate/validator/internal/engine/cascading/LegacyOptionalValueExtractor.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.engine.cascading;
+
+import java.util.Optional;
+
+import org.hibernate.validator.spi.cascading.ValueExtractor;
+
+/**
+ * Extractor for supporting the legacy construct {@code @Valid Optional<Foo>}.
+ * @author Gunnar Morling
+ *
+ */
+// TODO should we keep that, or only support {@code Optional<@Valid Foo>}.
```

Comment:
If we plan to remove it someday, I think the best is to remove it now.

But if it does not cost us, I think we should probably keep it (or throw an error if we consider it's not a valid pattern anymore)